test/examples.lean

```diff
@@ -206,6 +206,33 @@ meta structure meta_struct (α : Type u) : Type u :=
   (k : list (list α))
   (w : expr)
 
+@[derive [traversable,is_lawful_traversable]]
+inductive my_tree (α : Type)
+| leaf {} : my_tree
+| node : my_tree → my_tree → α → my_tree
+
+section
+open my_tree (hiding traverse)
+
+def x : my_tree (list nat) :=
+node
+  leaf
+  (node
+    (node leaf leaf [1,2,3])
+    leaf
+    [3,2])
+  [1]
+
+def ex : state (list ℕ) (my_tree $ list unit) :=
```

Comment:
Could you just add a comment here explaining in words what `ex` is meant to do? I'm failing to guess, even working both forwards from the definition and backwards from the example below. Ideally it would be clear to the reader that the example is showing the expected behaviour. :-)